File: content/remark/2020/03/currying/sample.js
// function add(x) {
//     return function(y) {
//         return x + y;
//     };
// }
const add = x => y => x + y;

console.log(add(1)(2)); //Output: 3
let increment = add(1);
console.log(increment(2)); //Output: 3

File: content/remark/2020/03/currying/sample.go
package main

import "fmt"

func add(x int) func(int) int {
	return func(y int) int {
		return x + y
	}
}

func main() {
	fmt.Println(add(1)(2)) //Output: 3
	increment := add(1)
	fmt.Println(increment(2)) //Output: 3
}
